src/pip/_internal/models/wheel.py

```diff
@@ -42,6 +43,9 @@ def __init__(self, filename):
         self.abis = wheel_info.group('abi').split('.')
         self.plats = wheel_info.group('plat').split('.')
 
+        # Adjust for any legacy custom PyPy tags found in pyversions
+        _add_standard_pypy_version_tags(self.pyversions)
```

Comment:
Like @chrahunt suggested, in case `_add_standard_pypy_version_tags` returns `True`, `pip` should output a `PipDeprecationWarning` since we won't keep this workaround forever.

Reply:
Adding the warning ended up having several ripple effects on the design of this helper function (most notably: making it side-effect free, and instead returning the list of new tags to be added to the `pyversions` sequence)